x/concentrated-liquidity/math/tick.go

```diff
@@ -120,55 +120,59 @@ func PriceToTick(price sdk.Dec) (sdk.Int, error) {
-	tickIndexModulus := tickIndex.Mod(tickSpacingInt)
-	if !tickIndexModulus.Equal(sdk.ZeroInt()) {
-		tickIndex = tickIndex.Sub(tickIndexModulus)
+	// Note that this is Euclidean modulus.
+	tickIndexModulus := tickIndex % int64(tickSpacing)
+	if tickIndexModulus < 0 {
+		tickIndexModulus += int64(tickSpacing)
+	}
+
+	if tickIndexModulus != 0 {
+		tickIndex = tickIndex - tickIndexModulus
 	}
 
 	// Defense-in-depth check to ensure that the tick index is within the authorized range
 	// Should never get here.
-	if tickIndex.GT(sdk.NewInt(types.MaxTick)) || tickIndex.LT(sdk.NewInt(types.MinTick)) {
-		return sdk.Int{}, types.TickIndexNotWithinBoundariesError{ActualTick: tickIndex.Int64(), MinTick: types.MinTick, MaxTick: types.MaxTick}
+	if tickIndex > types.MaxTick || tickIndex < types.MinTick {
+		return 0, types.TickIndexNotWithinBoundariesError{ActualTick: tickIndex, MinTick: types.MinTick, MaxTick: types.MaxTick}
 	}
 
 	return tickIndex, nil
 }
 
 // powTen treats negative exponents as 1/(10**|exponent|) instead of 10**-exponent
 // This is because the sdk.Dec.Power function does not support negative exponents
-func PowTenInternal(exponent sdk.Int) sdk.Dec {
-	if exponent.GTE(sdk.ZeroInt()) {
-		return sdkTenDec.Power(exponent.Uint64())
+func PowTenInternal(exponent int64) sdk.Dec {
+	if exponent > 0 {
+		return sdkTenDec.Power(uint64(exponent))
 	}
-	return sdk.OneDec().Quo(sdkTenDec.Power(exponent.Abs().Uint64()))
+	return sdk.OneDec().Quo(sdkTenDec.Power(uint64(exponent * -1)))
 }
 
-func powTenBigDec(exponent sdk.Int) osmomath.BigDec {
-	if exponent.GTE(sdk.ZeroInt()) {
-		return osmomath.NewBigDec(10).PowerInteger(exponent.Uint64())
+func powTenBigDec(exponent int64) osmomath.BigDec {
+	if exponent > 0 {
+		return osmomath.NewBigDec(10).PowerInteger(uint64(exponent))
 	}
-	return osmomath.OneDec().Quo(osmomath.NewBigDec(10).PowerInteger(exponent.Abs().Uint64()))
+	return osmomath.OneDec().Quo(osmomath.NewBigDec(10).PowerInteger(uint64(exponent * -1)))
 }
 
 // CalculatePriceToTick takes in a price and returns the corresponding tick index.
 // This function does not take into consideration tick spacing.
-func CalculatePriceToTick(price sdk.Dec) (tickIndex sdk.Int) {
+func CalculatePriceToTick(price sdk.Dec) (tickIndex int64) {
 	// The formula is as follows: geometricExponentIncrementDistanceInTicks = 9 * 10**(-exponentAtPriceOne)
 	// Due to sdk.Power restrictions, if the resulting power is negative, we take 9 * (1/10**exponentAtPriceOne)
 	exponentAtPriceOne := types.ExponentAtPriceOne
-	geometricExponentIncrementDistanceInTicks := sdkNineDec.Mul(PowTenInternal(exponentAtPriceOne.Neg()))
+	geometricExponentIncrementDistanceInTicks := sdkNineDec.Mul(PowTenInternal(exponentAtPriceOne * -1)).TruncateInt64()
 
 	// Initialize the current price to 1, the current precision to exponentAtPriceOne, and the number of ticks passed to 0
 	currentPrice := sdk.OneDec()
-	ticksPassed := sdk.ZeroInt()
+	ticksPassed := int64(0)
```

Comment:
I think we can just use 0 here instead of explicit type

Reply:
No, we cannot. Other functions operate on `int64` so we must have the types consistent